src/Sequence.cc

```diff
@@ -282,7 +284,34 @@ Sequence::ConstIterator Sequence::erase(Sequence::ConstIterator begin,
     return return_iter;
 }
 
-void Sequence::execute(Context& context, CommChannel* comm)
+void Sequence::execute(Context& context, CommChannel* comm_channel)
+{
+    handle_execution(context, comm_channel, "Sequence",
+        [this](Context& context, CommChannel* comm)
+        {
+            check_syntax();
+            execute_range(steps_.begin(), steps_.end(), context, comm);
+        });
+}
+
+void Sequence::execute_single_step(Context& context, CommChannel* comm_channel,
+                                   StepIndex step_index)
+{
+    if (step_index >= size())
+        throw Error(cat("Invalid step index ", step_index));
+
+    handle_execution(context, comm_channel, cat("Step ", step_index + 1),
```

Comment:
I must admit I was already uncomfortable when I wrote this. :)

There is no normal "step started" message anywhere, so AFAIK there can also be no confusion with other messages.  But the +1 problem is... hard. Maybe we should include the type/name of the step instead?
```
Single-step execution (ACTION "Insert screen") started
```

Reply:
> There is no normal "step started" message anywhere

But there is. `Step` sends a `Message` with `Step started`

![image](https://user-images.githubusercontent.com/16012374/208713601-86c02c50-f9a6-4d14-84bc-221083f6b499.png)

Reply:
... yes, but there is no related output. The message is not used anywhere. Maybe we should remove it.

Reply:
There is no related output?
The message is used nowhere?

But the `Message` is sent, or not?

I remember the first thing I did when I joined in to this taskomat stuff was writing a message-queue-dumper, so that one could see all messages and how the handling unfolds. Imho that is very instructive. That the Step starts and stops were clearly communicated what a big plus.

At the moment 'noone' is printing the messages. But for me they are vital execution details.
A user does not have to use the `Executor` when executing a `Sequence` in another context, Maybe someone wants to implement that with a thread pool ;-) or whatever. And for clarity outputs all messages into a log?

Also keep in mind #8, where textual representations are also a plus.

At least, I believe there is no `Message` without a textual message?
IF we drop the textual representation we should drop it for all `Message`s, i.e. also for _Sequence started_.

But possibly we would need then a `Message` dumper or stream-operator, that translates the messages to readable form. In fact you are right, we could move those texts to the (existing) Message-to-Stream op.
But then we would have it there, the `Step started` versus `Step 3 started` messages.

For the concrete 'issue' where this conversation started....
We have some Sequence-ish object that starts. I believe we should distinguish it from a Step (which is something different), it is a Mini-Sequence, a Single-Step-Sequence, or whatever. It 'holds' one Step, but it is not the Step. As also a Sequence with just one Step is not a Step, it's a something that is hierarchically more abstract. A Singlesteping.

Reply:
> There is no related output? The message is used nowhere?
> 
> But the `Message` is sent, or not?

Sorry, I was not precise. Of course the `Message` is sent and received, but the message *text* is in fact not used anywhere.
I think we should implement the message hooks you mentioned in #8 to enhance debuggability. For that, the message text may indeed be useful, so let's keep it.